### PR TITLE
ci: allow git fetch and git pull --rebase without permission prompts

### PR DESCRIPTION
**Pull request by scooter-dangle**
https://claude.ai/code/session_01FaUsuhhfUipqE17CoBMGoW

### CODE REVIEW

File: src/rotate.rs [hunk 1]
```diff
@@ -228,4 +228,66 @@ mod test_scan_row {
         r"        /         \        ",
         vec![(b')', 0)],
     ));
+
+    // All rows from rando_link rotated diagram (rows 0–9).
+
+    test!(rando_link_rotated_row_00(
+        r"        ____   _            ____    "
+        r"",
```
Comment (scooter-dangle) [inline src/rotate.rs]: For each of these new tests, the two input lines are listed in incorrect order (and should be swapped, @claude )

File: src/rotate.rs [hunk 2]
```diff
@@ -228,4 +228,66 @@ mod test_scan_row {
...
+    test!(rando_link_rotated_row_00(
+        r""
+        r"        ____   _            ____    ",
+        vec![],
+    ));
+
+    test!(rando_link_rotated_row_01(
+        r"        ____   _            ____    "
+        r"       /    \ / \          /    \   ",
+        vec![],
```
Comment (scooter-dangle) [inline src/rotate.rs]:
```suggestion
        vec![(b')', 4), (b')', 2), (b')', 0)],
```

File: src/rotate.rs [hunk 3]
```diff
@@ -228,4 +228,66 @@ mod test_scan_row {
...
+    test!(rando_link_rotated_row_02(
+        r"       /    \ / \          /    \   "
+        r"      /      \   \        /      )  ",
+        vec![],
+    ));
+
+    test!(rando_link_rotated_row_03(
+        r"      /      \   \        /      )  "
+        r"     /   ___/ \   \      /  _   /   ",
+        vec![],
```
Comment (scooter-dangle) [inline src/rotate.rs]:
```suggestion
        vec![(b'/', 3)],
```

File: src/rotate.rs [hunk 4]
```diff
@@ -228,4 +228,66 @@ mod test_scan_row {
...
+    ));
+
+    test!(rando_link_rotated_row_04(
+        r"     /   ___/ \   \      /  _   /   "
+        r"    /   /      \   \    /  / \ /    ",
+        vec![],
```
Comment (scooter-dangle) [inline src/rotate.rs]:
```suggestion
        vec![(b')', 1)],
```

File: src/rotate.rs [hunk 5]
```diff
@@ -228,4 +228,66 @@ mod test_scan_row {
...
+    test!(rando_link_rotated_row_05(
+        r"    /   /      \   \    /  / \ /    "
+        r"   /   (        )   \  /  /   \     ",
+        vec![],
+    ));
+
+    test!(rando_link_rotated_row_06(
+        r"   /   (        )   \  /  /   \     "
+        r"  /     \   ___/     \/  /   / \__  ",
+        vec![],
```
Comment (scooter-dangle) [inline src/rotate.rs]:
```suggestion
        vec![(b'/', 0)],
```

File: src/rotate.rs [hunk 6]
```diff
@@ -228,4 +228,66 @@ mod test_scan_row {
...
+    ));
+
+    test!(rando_link_rotated_row_07(
+        r"  /     \   ___/     \/  /   / \__  "
+        r" /       \ /            /   /     \ ",
+        vec![],
```
Comment (scooter-dangle) [inline src/rotate.rs]:
```suggestion
        vec![(b'(', 3)],
```

File: src/rotate.rs [hunk 7]
```diff
@@ -228,4 +228,66 @@ mod test_scan_row {
...
+    test!(rando_link_rotated_row_08(
+        r" /       \ /            /   /     \ "
+        r"(         \            /   (       )",
+        vec![],
+    ));
+
+    test!(rando_link_rotated_row_09(
+        r"(         \            /   (       )"
+        r" \_______/ \__________/     \_____/ ",
+        vec![],
```
Comment (scooter-dangle) [inline src/rotate.rs]:
```suggestion
        vec![(b'/', 3)],
```

File: src/rotate.rs [hunk 8]
```diff
@@ -228,4 +228,66 @@ mod test_scan_row {
...
+
+    test!(rando_link_rotated_row_08(
+        r" /       \ /            /   /     \ "
+        r"(         \            /   (       )",
+        vec![],
+    ));
+
+    test!(rando_link_rotated_row_09(
+        r"(         \            /   (       )"
+        r" \_______/ \__________/     \_____/ ",
```
Comment (scooter-dangle) [inline src/rotate.rs:290]: @claude , add one more test case where this is the initial input row and the next input row is just an empty string

File: src/rotate.rs [hunk 9]
```diff
@@ -228,4 +228,72 @@ mod test_scan_row {
...
+        vec![(b'/', 3)],
+    ));
+
+    test!(rando_link_rotated_row_10(
+        r" \_______/ \__________/     \_____/ "
+        r"",
+        vec![(b'(', 0)],
```
Comment (scooter-dangle) [inline src/rotate.rs]:
```suggestion
        vec![(b'(', 0), (b'(', 2), (b'(', 4)],
```
